### PR TITLE
Support for Tinyboy/Turnigy Fabrikator Mini 1.5

### PR DESCRIPTION
**Pull request by reibuehl**
Machine and extruder definitions and print profiles for normal, high quality and draft printing for the Tinyboy Fabrikator Mini 1.5 (also known as Turnigy/HobbyKing Fabrikator Mini 1.5.

### CODE REVIEW

**Comment by konskarm** [comment]
Hi @reibuehl. Please update the "setting_version" from 15 to 16 in the metadata of the qualities.

**Comment by reibuehl** [comment]
I changed the "setting_version" from 15 to 16 in the metadata of the qualities as requested.

**Comment by konskarm** [comment]
@reibuehl  Cura complains about the following:
```
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'fill_density' placeholder in start/end g-code
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'print_time' placeholder in start/end g-code
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'filament_amount' placeholder in start/end g-code
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'filament_weight' placeholder in start/end g-code
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'filament_cost' placeholder in start/end g-code
WARNING - [Thread-7] CuraEngineBackend.StartSliceJob.get_value [85]: Unable to replace 'profile_string' placeholder in start/end g-code
```
Are they indeed necessary for the start/end gcodes? Cura will definitely not fill them in.

**Comment by reibuehl** [comment]
No, these fields are not needed. I think they are still remains from a UM2 Cura 15.04 header that was used as a template to create this start and end code. From looking at https://github.com/Ultimaker/Cura/blob/master/resources/definitions/fdmprinter.def.json I would think that infill_sparse_density is the current equivalent of fill_density. I couldn't find any equivalents for the cost calculation and profile name fields yet. Has the current Cura completely dropped the support for these variables? Then I can remove them. Or have they also just been renamed? Then I can also change them if somebody can point me to the file where the current ones are defined/described.

File: resources/definitions/tinyboy_fabrikator15.def.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        },
+        "machine_end_gcode": {
+            "default_value": ";End GCode\nM104 S0 ;extruder heater off\nM140 S0 ;heated bed heater off (if you have it)\nG91 ;relative positioning\nG1 E-1 F300 ;retract the filament a bit before lifting the nozzle, to release some of the pressure\nG1 Z+0.5 E-5 X-20 Y-20 F{travel_speed} ;move Z up a bit and retract filament even more\nG28 X0 Y0 ;move X/Y to min endstops, so the head is out of the way\nM84 ;steppers off\nG90 ;absolute positioning\n;{profile_string}"
```
Comment (konskarm) [inline resources/definitions/tinyboy_fabrikator15.def.json:39]:
```suggestion
            "default_value": ";End GCode\nM104 S0 ;extruder heater off\nM140 S0 ;heated bed heater off (if you have it)\nG91 ;relative positioning\nG1 E-1 F300 ;retract the filament a bit before lifting the nozzle, to release some of the pressure\nG1 Z+0.5 E-5 X-20 Y-20 F{travel_speed} ;move Z up a bit and retract filament even more\nG28 X0 Y0 ;move X/Y to min endstops, so the head is out of the way\nM84 ;steppers off\nG90 ;absolute positioning\n;{jobname}"

```

**Review by konskarm — COMMENTED** [review]
It turns out that these two need to be changed for this to work. Also, the warning message is generated before the slice, when the filament amount, weight and cost information is not available yet, but they are properly filled once the slice is completed, so there is no issue there.

**Comment by Ghostkeeper** [comment]
The build plate mesh isn't aligned properly. Models are placed inside the build plate mesh.
image
I can fix that up for you.

**Comment by Ghostkeeper** [comment]
For the rest it looks okay.
You're making Z moves using `G1 F{travel_speed}` in the start g-code. I'm going to assume that this is correctly moving at 100mm/min, not 100mm/s. 100mm/min (1.67mm/s) is on the slow side for a Z move but still okay. 100mm/s is extremely fast for Z moves unless it's a delta printer.

**Review by Ghostkeeper — APPROVED** [review]

**Comment by reibuehl** [comment]
> The build plate mesh isn't aligned properly. Models are placed inside the build plate mesh.
> I can fix that up for you.

Yes, please do so. Thanks!

**Comment by Ghostkeeper** [comment]
It was corrected here: https://github.com/Ultimaker/Cura/commit/f78fd9b3a44314729f5c1980f59c89310608bd9e

**Pull request merged**